File: win-prog/ProgWinCS/Font Fun/BaselineTilt/BaselineTilt.cs
//-------------------------------------------
// BaselineTilt.cs © 2001 by Charles Petzold
//-------------------------------------------
using System;
using System.Drawing;
using System.Drawing.Drawing2D;
using System.Windows.Forms;

class BaselineTilt: FontMenuForm
{
     public new static void Main()
     {
          Application.Run(new BaselineTilt());
     }
     public BaselineTilt()
     {
          Text = "Baseline Tilt";

          strText = "Baseline";
          font = new Font("Times New Roman", 144);
     }
     protected override void DoPage(Graphics grfx, Color clr, int cx, int cy)
     {
          float yBaseline = 3 * cy / 4;
          float cyAscent = GetAscent(grfx, font);

          grfx.DrawLine(new Pen(clr), 0, yBaseline, cx, yBaseline);

          grfx.TranslateTransform(0, yBaseline);

          Matrix matx = grfx.Transform;
          matx.Shear(-0.5f, 0);
          grfx.Transform = matx;

          grfx.DrawString(strText, font, new SolidBrush(clr), 0, -cyAscent);
     }
}

File: win-prog/ProgWinVB/Font Fun/BaselineTilt/BaselineTilt.vb
'---------------------------------------------
' BaselineTilt.vb (c) 2002 by Charles Petzold
'---------------------------------------------
Imports System
Imports System.Drawing
Imports System.Drawing.Drawing2D
Imports System.Windows.Forms

Class BaselineTilt
    Inherits FontMenuForm

    Shared Shadows Sub Main()
        Application.Run(New BaselineTilt())
    End Sub

    Sub New()
        Text = "Baseline Tilt"
        strText = "Baseline"
        fnt = New Font("Times New Roman", 144)
    End Sub

    Protected Overrides Sub DoPage(ByVal grfx As Graphics, _
            ByVal clr As Color, ByVal cx As Integer, ByVal cy As Integer)
        Dim yBaseline As Single = 3 * cy \ 4
        Dim cyAscent As Single = GetAscent(grfx, fnt)

        grfx.DrawLine(New Pen(clr), 0, yBaseline, cx, yBaseline)
        grfx.TranslateTransform(0, yBaseline)

        Dim matx As Matrix = grfx.Transform
        matx.Shear(-0.5, 0)
        grfx.Transform = matx

        grfx.DrawString(strText, fnt, New SolidBrush(clr), 0, -cyAscent)
    End Sub
End Class
